packages/core/src/MachineNode.ts

```diff
@@ -273,7 +255,25 @@ export class MachineNode<
     >;
   }
 
-  public getStateNodeById(id: string): StateNode<TContext, TEvent> {
-    return getStateNodeById(this, id);
+  public getStateNodeById(stateId: string): StateNode<TContext, TEvent> {
```

Comment:
does this new implementation of this method result in a different behavior somehow?

Reply:
It does not. `getStateNodeById` should only be used for the entire machine anyway.

Reply:
Hm, I feel like I'm missing something - there is still a standalone `getStateNodeById` util that is used in some places in the code. If this method here is the equivalent of that util (which was used here before this PR) then shouldn't that util be removed and calls to it should be replaced with calls to this method?

Reply:
Removed in ba8ea1a